mdbook-tera-backend/src/tera_renderer/renderer.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{anyhow, Result};
+use mdbook::renderer::RenderContext;
+use serde_json::to_value;
+use std::collections::{BTreeMap, HashMap};
+use std::fs;
+use std::io::Write;
+use std::path::Path;
+use std::sync::Arc;
+use tera::Tera;
+
+/// Renderer for the tera backend.
+///
+/// This will read all the files in the `RenderContext` and render them using the `Tera` template.
+///
+/// # Example
+///
+/// ```
+/// let mut stdin = io::stdin();
+/// let ctx = RenderContext::from_json(&mut stdin).unwrap();
+/// let config: TeraRendererConfig = ctx
+///     .config
+///     .get_deserialized_opt("output.tera-backend")
+///     .expect("Failed to get tera-backend config")
+///     .unwrap();
+///
+/// let tera_template = config
+///    .create_template(&ctx.root)
+///    .expect("Failed to create components");
+/// let mut renderer = Renderer::new(ctx, tera_template).expect("Failed to create renderer");
+/// renderer.render_book().expect("Failed to render book");
+/// ```
+pub(crate) struct Renderer {
+    ctx: Arc<RenderContext>,
+    tera_template: Tera,
+}
+
+impl Renderer {
+    /// Create a new `Renderer` from the `RenderContext` and `Tera` template.
+    pub(crate) fn new(ctx: RenderContext, tera_template: Tera) -> Result<Renderer> {
+        let mut renderer = Renderer {
+            ctx: Arc::new(ctx),
+            tera_template,
+        };
+        renderer
+            .tera_template
+            .register_function("get_context", renderer.create_get_context_function());
+        Ok(renderer)
+    }
+
+    /// Render the book. This goes through the output of the HTML renderer
+    /// by considering all the output HTML files as input to the Tera template.
+    /// It overwrites the preexisting files with their Tera-rendered version.
+    pub(crate) fn render_book(&mut self) -> Result<()> {
+        let dest_dir = self
+            .ctx
+            .destination
+            .parent()
+            .unwrap()
+            .join("html")
+            .to_owned();
```

Comment:
No `to_owned` needed, `join` already return you an owned value:

```suggestion
        let dest_dir = self.ctx.destination.parent().unwrap().join("html");
```